qiskit/circuit/quantumcircuitdata.py

```diff
@@ -192,7 +193,7 @@ def _resolve_legacy_value(self, operation, qargs, cargs) -> CircuitInstruction:
         return CircuitInstruction(operation, tuple(qargs), tuple(cargs))
 
     def insert(self, index, value):
-        self._circuit._data.insert(index, None)
+        self._circuit._data.insert(index, CircuitInstruction(None, [], []))
         try:
```

Comment:
`CircuitInstruction` wants tuples not lists - it's pretty minor, but we may as well pass the right thing immediately (`tuple(t)` when `t` is a tuple is a no-op)